### ISSUE TITLE
Impossible to add getter to populated virtual

### ISSUE THREAD

**Issue by jeroenvollenbrock** [Automattic/mongoose#5835]
**Do you want to request a *feature* or report a *bug*?**
Both

**What is the current behavior?**
When using a populated virtual with an added getter, the added getter is called before the populating getter due to the reverse-looping over the getter array (https://github.com/Automattic/mongoose/blob/6535373788fc16f0613e3d3e40fb01d77f981c24/lib/virtualtype.js#L76)

**If the current behavior is a bug, please provide the steps to reproduce.**
```
let defaultUser = {
    _id: '123',
    name: 'test'
};

let schema = <schema>;
schema.virtual('user', {
    ref: 'User'
    localField: 'userId',
    foreignField: '_id',
    justOne: true,
}).get(function(user) {
    return user || defaultUser;
});
```

Using the non-documented `getters` property via `virtual.getters.unshift(<function>);` works

**What is the expected behavior?**
The virtual user should resolve to the specified user or the default object. Perhaps it would be a good idea to either specify a default value in the virtual opts, or provide a public function to prepend getters to virtuals in order to prevent breaking API changes. Please note that we'd also like to be able to specify `undefined` as a default value as well in some cases, so the entire field gets excluded in JSON responses if it is not populated, instead of setting it to null.

**Comment by vkarpov15** [comment]
Yeah I agree the reverse getter order is a nightmare. There was another issue related to this, #4832. Please follow #4835 for updates on this issue, we're planning on making a 5.0 release within the next few months. You already found the `virtual.getters.unshift()` workaround so please just use that for now :+1:

**Comment by Karabur** [comment]
#4835 is fixed but it fixes only `Schema.prototype.applyGetters`, but not `VitrualSchema.prototype.applyGetters`, which apparently has same code and still in reverse order. 
Please consider reopening issue as it still not fixed for case described here.

**Comment by vkarpov15** [comment]
Fixed by #8897
